### ISSUE TITLE
routing

### ISSUE THREAD

**Issue by rockneverdies55** [gopangolin/wp-reactivate#7]
Hi there,

Sorry if I'm asking something obvious. I'm a newbie to working with wordpress.

There is nothing about routing in the documentation or setup. Can you guys elaborate on that a bit?

For example let's say my plugin is loaded at `/myplugin` (`http://my-site-url.com/myplugin`) by wordpress. And the sub-routes of `/myplugin` is handled by react-router e.g. `/myplugin/myroute1` (`http://site-url.com/myplugin/myroute1`) or `/myplugin/myroute2` (`http://site-url.com/myplugin/myroute2`) etc which works fine (for me without an issue using react-router-dom project).

However, wordpress yields "OOPS! THAT PAGE CAN’T BE FOUND." when I try to load the site directly with a subroute in the address bar e.g. `http://site-url.com/myplugin/myroute2`.

Thanks.

**Comment by dnusca** [comment]
Hey,

Could elaborate on your use case? Are you rendering a sort of front end app into a WP theme?

I would suggest using [query parameters](https://reacttraining.com/react-router/web/example/query-parameters) so as not to interfere with the WP routing.

**Comment by rockneverdies55** [comment]
I created a shortcode using this repo boilerplate which will render full spa. We'd like user of our plugin to be able to add our shortcode to any page in the site; and our plugin should be able to handle all the subroutes from there on. For example, user creates `/vehicles` page on his/her wordpress site and wants our plugin to render the content in that page; drops our shortcode to the page and shortcode will handle all the routing from /vehicles route such as `/vehicles/personal`, `/vehicles/commercial/trucks/18-tires`, etc... The requirement is that the urls should be nice looking html5 style urls means no hashes or query params...

To accomplish that, I've done:

Within the [shortcode function](https://github.com/gopangolin/wp-reactivate/blob/master/includes/Shortcode.php#L87) just before the line-87, I added a function call such as 
```
global $post;
$this->spa_url_rewrite($post->ID, $post->post_name);
```
and within `spa_url_rewrite` function I have:

```
	function spa_url_rewrite($postID, $postName) {
		global $wp_rewrite;
		$existingRules = $wp_rewrite->rules;
		$spaRegex = '^' . $postName . '\/(.*)\/?$';
		if(!array_key_exists($spaRegex , $existingRules)) {
			add_rewrite_rule( $spaRegex, "index.php?page_id={$postID}", 'top' );
			$wp_rewrite->flush_rules();
		}
	}

```

I tried to mitigate the cost of flushing rules on each page load by checking if existing rules already got my rule added before flushing. But not sure whether there is a better way to do this or not. My solution needs to be dynamic though. I mean I can't blindly add my overwrite rule to /vehicles route for example on init hook. My plugin should work whatever page end user has added my shortcode into.
